src/main/java/de/chojo/repbot/dao/access/guild/Gdpr.java

```diff
@@ -17,25 +21,36 @@ public Guild guild() {
         return repGuild.guild();
     }
 
+    @Override
+    public long guildId() {
+        return repGuild.guildId();
+    }
+
     public void queueDeletion() {
-        builder()
+        if (builder()
                 .query("""
                        INSERT INTO
-                           cleanup_schedule(guild_id)
-                           VALUES (?)
+                           cleanup_schedule(guild_id, user_id)
+                           VALUES (?, 0)
                                ON CONFLICT(guild_id, user_id)
                                    DO NOTHING;
                        """)
                 .parameter(stmt -> stmt.setLong(guildId()))
                 .update()
-                .sendSync();
+                .sendSync()
+                .changed()) {
+            log.debug("Queuing guild {} for deletion.", guildId());
+        }
     }
 
     public void dequeueDeletion() {
-        builder()
-                .query("DELETE FROM cleanup_schedule WHERE guild_id = ?;")
+        if (builder()
+                .query("DELETE FROM cleanup_schedule WHERE guild_id = ? AND user_id = 0;")
```

Comment:
Currently when the bot left and rejoined a guild during grace period all scheduled jobs for the guild were removed including user data removal, which was not intended.